### DISCUSSION TITLE
How to separate classes in different compile units?

### DISCUSSION THREAD

**Discussion by EdoPro98** [wjakob/nanobind#35] [Q&A] [answered]
I am trying to port my library [SimSiPM](https://github.com/EdoPro98/SimSiPM/tree/dev) from Pybind11 to nanobind. I am able to compile it if i put all class definitions in a single wrapper file. Nevertheless I find much more elegant to separate each class wrapper in a different file and call all the methods for wrapping in a "main" file as I used to do in Pybind11. Also in Pybind11 documentation it is stated that this should speed up compilation time I guess due to parallelization of compile jobs.

The part that is blocking me in doing so is that the nanobind corresponding class of py::module is missing.
For example in Pybind11 I did the following:
```cpp
// #inlcude...

void SiPMPropertiesPy(py::module&);
void SiPMAnalogSignalPy(py::module&);
void SiPMDigitalSignalPy(py::module&);
void SiPMDebugInfoPy(py::module&);
void SiPMHitPy(py::module&);
void SiPMSensorPy(py::module&);
void SiPMRandomPy(py::module&);

PYBIND11_MODULE(SiPM, m) {
  m.doc() = "Module for SiPM simulation";
  m.attr("__version__") = SIPM_VERSION;
  SiPMPropertiesPy(m);
  SiPMAnalogSignalPy(m);
  SiPMDigitalSignalPy(m);
  SiPMDebugInfoPy(m);
  SiPMHitPy(m);
  SiPMSensorPy(m);
  SiPMRandomPy(m);
}
```
Where each of the `void ClassPy(py::module&)` is the corresponding wrapper of each class in the module.
For the moment I have been testing nanobind by putting everything in one file and it works properly.

Am I missing something here?

**Comment by ubt0001** [comment, marked as answer]
maybe ```void ClassPy(nb::module_&)```?

**Reply by wjakob** [reply]
Exactly. This should work exactly the same way @EdoPro98, you just have to use ``nb::module_``. This was renamed since ``module`` is a special keyword in C++20+.
